### PR TITLE
Add I2C slave support for nRF

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

This builds on top of https://github.com/tock/tock/pull/2612 by adding nRF I2C `master_slave_driver` support.

### Testing Strategy

Running I2C communication on the nRF

### TODO or Help Wanted

Nothing

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/i2c_master_slave_driver.rs [hunk 1]
```diff
@@ -130,7 +130,7 @@ impl hil::i2c::I2CHwMasterClient for I2CMasterSlaveDriver<'_> {
                         // than the buffer could lead to array overrun errors in
                         // userspace. The I2C syscall API should pass back lengths.
                         // -pal 3/5/21
-                        app.master_tx_buffer.mut_map_or(0, move |app_buffer| {
+                        app.master_rx_buffer.mut_map_or(0, move |app_buffer| {
```
Comment (hudson-ayers) [inline capsules/src/i2c_master_slave_driver.rs:133]: Was this a bug that it used `master_tx_buffer` before? Or is the idea that `WriteRead` ops should send from the tx buffer but read into the rx buffer?

Should this be accompanied by a TRD update or at least a libtock-c update to reflect which buffer is used here?

**Review by hudson-ayers — APPROVED** [review]
Looks good, though one question about a buffer change in `capsules`

File: capsules/src/i2c_master_slave_driver.rs (hunk 1 above)
Comment (alistair23) [reply capsules/src/i2c_master_slave_driver.rs:133]: The libtock-c documentation and interface isn't clear, but this seems like a bug.

The `MasterAction::Read` will copy the read data to the `app.master_rx_buffer` buffer while the read action from `MasterAction::WriteRead` copies data to `app.master_tx_buffer`. This changes all read operations to store data to `app.master_rx_buffer`. I'll add a comment to the libtock-c interface to make this clear

Comment (hudson-ayers) [reply capsules/src/i2c_master_slave_driver.rs:133]: That sounds right to me, thanks

**Comment by alistair23** [comment]
This is ready to go

**Review by bradjc — APPROVED** [review]

**Pull request merged**